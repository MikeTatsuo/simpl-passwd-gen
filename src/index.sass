@import styles/utils
@import styles/base
@import styles/themes
@import styles/layout
