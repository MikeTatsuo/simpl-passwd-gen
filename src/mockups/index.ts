export * from './Checkbox'
export * from './Header';
