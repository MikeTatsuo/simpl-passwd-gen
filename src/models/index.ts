export * from './CardFactoryTypes.enum';
export * from './CardProps.interface';
export * from './CheckboxProps.interface';
export * from './HeaderProps.interface';
