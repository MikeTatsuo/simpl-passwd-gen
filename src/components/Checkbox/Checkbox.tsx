import { Component, SyntheticEvent } from 'react';

import { CheckboxProps } from 'src/models';

export default class Checkbox extends Component<CheckboxProps> {
	constructor(props: CheckboxProps) {
		super(props);

		this.checkChange = this.checkChange.bind(this);
	}

	checkChange(e: SyntheticEvent) {
		(this.props.onCheckChange as (value: boolean, name: string) => void)(
			(e.target as HTMLInputElement).checked,
			this.props.name ?? ''
		);
	}

	render() {
		return (
			<div className="checkbox">
				<input
					className="checkbox__input"
					type="checkbox"
					name={this.props.name ?? ''}
					checked={this.props.checked ?? false}
					onChange={this.checkChange}
				/>
				<label htmlFor="checkbox__label">{this.props.label ?? ''}</label>
			</div>
		);
	}
}
