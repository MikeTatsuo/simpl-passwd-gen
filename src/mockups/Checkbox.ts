import { datatype, lorem } from 'faker';

import { CheckboxProps } from "src/models";

export const CheckboxMockup: CheckboxProps = {
	checked: datatype.boolean(),
	label: lorem.word(),
	name: lorem.word(),
};
