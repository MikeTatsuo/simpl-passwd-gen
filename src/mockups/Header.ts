import { lorem } from 'faker';

import { HeaderProps } from 'src/models';

export const HeaderMockup: HeaderProps = {
	title: lorem.sentence(),
}
