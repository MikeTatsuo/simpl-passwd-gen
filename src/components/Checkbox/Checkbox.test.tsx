import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';

import Checkbox from 'src/components/Checkbox/Checkbox';

import { CheckboxMockup } from 'src/mockups';

const { label, name, checked } = CheckboxMockup;
let container: HTMLDivElement;

describe('Checkbox.tsx', () => {
	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		unmountComponentAtNode(container);
		container.remove();
	});

	describe('snapshot', () => {
		it('should match snapshot', () => {
			act(() => {
				render(<Checkbox label="label" name="name" />, container);
			});

			expect(container).toMatchSnapshot();
		});
	});

	describe('data', () => {
		it('should display label', () => {
			act(() => {
				render(<Checkbox label={label} />, container);
			});

			expect(container.textContent).toMatch(label as string);
		});

		it('should render name', () => {
			act(() => {
				render(<Checkbox name={name} />, container);
			});

			expect(container.querySelector('input')?.getAttribute('name')).toMatch(
				name as string
			);
		});

		it('should not be checked', () => {
			act(() => {
				render(<Checkbox checked={false} />, container);
			});

			expect(container.querySelector('input')).not.toBeChecked();
		});

		it('should be checked', () => {
			act(() => {
				render(<Checkbox checked={true} />, container);
			});

			expect(container.querySelector('input')).toBeChecked();
		});
	});

	describe('onChange', () => {
		it('should dispatch onCheckChange event', () => {
			const onCheck = jest.fn();

			act(() => {
				render(
					<Checkbox
						label={label}
						name={name}
						checked={checked}
						onCheckChange={onCheck}
					/>,
					container
				);
			});

			const input = container.querySelector('input');
			act(() => {
				input?.dispatchEvent(new MouseEvent('click', { bubbles: true }));
			});

			expect(onCheck).toHaveBeenCalled();
			expect(onCheck).toHaveBeenCalledTimes(1);
			expect(onCheck).toHaveBeenCalledWith(!checked, name);
		});
	});
});
