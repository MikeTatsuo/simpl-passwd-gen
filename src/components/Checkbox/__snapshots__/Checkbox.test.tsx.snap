// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox.tsx snapshot should match snapshot 1`] = `
<div>
  <div
    class="checkbox"
  >
    <input
      class="checkbox__input"
      name="name"
      type="checkbox"
    />
    <label
      for="checkbox__label"
    >
      label
    </label>
  </div>
</div>
`;
